libutils/string_lib.c

```diff
@@ -372,13 +372,140 @@ bool EmptyString(const char *s)
+    assert(ERANGE != 0);
+
+    char *endptr = NULL;
+    long val;
+
+    errno = 0;
+    val = strtol(str, &endptr, 10);
+
+    if ((errno == ERANGE && (val == LONG_MAX || val == LONG_MIN)))
+    {
+        return ERANGE; // Overflow or underflow
+    }
+
+    if (endptr == str)
+    {
+        return -81; // No digits found
+    }
+
+    if (endptr == NULL)
+    {
+        return -82; // endpointer not set by strtol
+    }
+
+    if (*endptr != '\0' && !isspace(*endptr))
+    {
+        return -83; // string not properly terminated
+    }
+
+    if (errno != 0)
+    {
+        return errno; // Unknown error
+    }
+
+    *number = val;
+    return 0;
+}
+
+/**
+ * @brief Log a conversion error with an identifier for debugging
+ */
+void LogConversionError(const char *str_attempted, const char *id)
+{
+    Log(LOG_LEVEL_ERR, "Conversion error on '%s' (%s)", str_attempted, id);
+}
+
+/**
+ * @brief Converts a string of numerals in base 10 to a long integer,
+ *        uses a default value if errors occur.
+ *
+ * @see StringToLong()
+ * @param[in] str String with numerals to convert, cannot be NULL
+ * @param[in] default_return Value to return on error
+ * @return Result of conversion (or default_return in case of error)
+ */
+long StringToLongDefaultOnError(const char *str, long default_return)
+{
+    assert(str != NULL);
+    long result = 0;
+    int return_code = StringToLong(str, &result);
+    if (return_code != 0)
+    {
+        // Do not log anything because this can be used frequently
+        return default_return;
+    }
+    return result;
+}
+
+/**
+ * @brief Converts a string of numerals in base 10 to a long integer, exits on error.
+ *
+ * Only use this function in contexts/components where it is acceptable
+ * to immediately exit when something goes wrong.
+ *
+ * @warning This function can exit the process based on string contents
+ * @see StringToLong()
+ * @param[in] str String with numerals to convert, cannot be NULL
+ * @return Result of conversion
+ */
+long StringToLongExitOnError(const char *str)
+{
+    assert(str != NULL);
+    long result;
+    int return_code = StringToLong(str, &result);
+    if (return_code != 0)
+    {
+        LogConversionError(str, "StringToLongExitOnError");
+        if (return_code == ERANGE)
+        {
+            Log(LOG_LEVEL_ERR, "Internal string to number conversion failed (overflow)");
+        }
+        else
+        {
+            Log(LOG_LEVEL_ERR, "Internal string to number conversion failed (%d)", return_code);
+        }
+        exit(EXIT_FAILURE);
+    }
+    return result;
+}
+
+/**
+ * @brief Convert a string of numerals to a long integer (deprecated).
+ *
+ * @warning This function is deprecated, do not use it
+ * @warning This function can exit the process based on string contents
+ * @see StringToLongExitOnError()
+ * @param[in] str String with numerals to convert, cannot be NULL
+ * @return Result of conversion
+ */
+long StringToLongUnsafe(const char *str)
```

Comment:
I thought you replaced this everywhere? Or where there too many occurences after all?

Reply:
If you think all of the places where this function has been replaced with new functions (with slightly differing behavior) are good, we can remove the old version.

Reply:
Since you are triggering builds like crazy, why don't you add a commit that removes it and trigger another, to see if/what breaks? :-)

Reply:
Nothing will break if I remove it, it is unused.

I am asking you to review the places where I've made changes by replacing the old StringToLong with new, different, functions.

Reply:
There are very few places in this pull request and I checked them, they look OK.